ray-operator/controllers/ray/common/service.go

```diff
@@ -38,6 +38,26 @@ func BuildServiceForHeadPod(cluster rayiov1alpha1.RayCluster) (*corev1.Service,
 	return service, nil
 }
 
+// BuildServiceForRayService Builds the service for a pod. Currently, there is only one service that allows
+// the worker nodes to connect to the head node.
+// RayService controller updates the service whenever a new RayCluster serves the traffic.
+func BuildServiceForRayService(rayService rayiov1alpha1.RayService, rayCluster rayiov1alpha1.RayCluster) (*corev1.Service, error) {
```

Comment:
I see, we use rayService.Name when building for the RayService service and rayCluster.Name for the RayCluster service...which should avoid collisions.

That seems ok...we'll have to make sure to document the distinction as it could get a little confusing.

It could theoretically make sense for the RayService service to only expose the serve port.
(rather than everything: serve, client, gcs)
@simon-mo you might have more context

On the other hand, the RayCluster controller should definitely create a service exposing the gcs-server port (which allows the workers to connect to head)

I'm probably overthinking it.